libs/langchain/langchain/vectorstores/opensearch_vector_search.py

```diff
@@ -511,8 +573,16 @@ def _raw_similarity_search_with_score(
                     vector_field=vector_field,
                     subquery_clause=subquery_clause,
                 )
+            elif efficient_filter != {}:
+                search_query = _approximate_search_query_with_efficient_filter(
+                    embedding, efficient_filter, k=k, vector_field=vector_field
+                )
             elif lucene_filter != {}:
-                search_query = _approximate_search_query_with_lucene_filter(
+                warnings.warn(
+                    "`lucene_filter` is deprecated. Please use the keyword argument"
```

Comment:
just checking, this applies not just for aoss right?

Reply:
This is generic change applicable for all of them, not just for AOSS